Fixed issue #425 (broken count)

SimplyComple0x78 opened the pull request:
# Description

I fixed the issue #425 but I don't know if there is another reason for that issue to happen.
Please investigate that in the review.

# Issue

https://github.com/neo4j/graphql/issues/425

# Checklist

The following requirements should have been met (depending on the changes in the branch):

- [ ] Documentation has been updated
- [ ] TCK tests have been updated
- [ ] Integration tests have been updated
- [ ] Example applications have been updated
- [ ] New files have copyright header
- [ ] CLA (https://neo4j.com/developer/cla/) has been signed

darrellwarde commented on packages/graphql/src/schema/resolvers/count.ts:
```diff
@@ -34,7 +34,12 @@ export default function countResolver({ node }: { node: Node }) {
         });
 
-        return result.records[0]._fields[0].toNumber();
+        const count: number | any = result.records[0]._fields[0];
+
+        if (typeof count === "number") {
+            return count;
+        }
+        return count.toNumber();
```
Damn, this is definitely needed here, sorry about that! We actually do this slightly differently elsewhere in the codebase and would like to stick with that for consistency:

```suggestion
        const count = result.records[0]._fields[0];

        // @ts-ignore: count is unknown, and to cast to object would be an antipattern
        if (isInt(count)) {
            return (count as Integer).toNumber();
        }

        return count;
```

Which will also require the following imports at the top of the file:

```ts
import { isInt, Integer } from "neo4j-driver";
```

SimplyComple0x78 replied:
Agreed, changed that part.

darrellwarde reviewed (changes requested):
Thanks for the contribution!

Before we can do anything with this, can you please [sign our CLA](https://neo4j.com/developer/cla/#sign-cla)?

See my comment regarding changes to be made. Would also be good to see the functionality tested for both a plain number and Neo4j Integer.

SimplyComple0x78 commented:
Changed the implementation as requested, added unit-tests for both cases and signed the CLA.

Is there anything left?
* Sadly, the tests weren't successful even after my changes, so I only checked the tests my new ones are in.
* I don't know how you're working with mocking - I felt the need to mock `import { execute } from "../../utils";` and did so (I don't know what's happening in there). In case that's the wrong direction, feel free to point me on other options.

darrellwarde commented:
Morning! There's a little manual step on our side for the CLA, so I've just done that now and will re-run the tests shortly. 🙂 

I actually see that there are no comprehensive unit tests for any of the resolvers... So maybe don't worry so much. Was just taking a very much "belt and braces" approach!!!

Regarding TeamCity, it's something we're looking to address by switching over to GitHub Actions so that open-source contributors can have better access to our pipelines. 🙂

danstarns commented:
I do find it really odd how and when this would not be an Integer. Seeming we simply return the result of executing [count](https://neo4j.com/docs/cypher-manual/current/functions/aggregating/#functions-count) that happens to return an Integer.

darrellwarde commented:
> I do find it really odd how and when this would not be an Integer. Seeming we simply return the result of executing [count](https://neo4j.com/docs/cypher-manual/current/functions/aggregating/#functions-count) that happens to return an Integer.

I think it can end up being either a native `number` or driver `Integer`, as we've experienced (painfully) throughout the rest of the project - we just forgot to put the safeguard here!

darrellwarde reviewed (approved):
The tests are exactly what I was looking for, thanks for this fix! 🙂
